Update proxy as Slice type (#401)
icerpc/icerpc-docs: content/slice/language-guide/interface.md

implement the C# (or Rust, Python...) abstractions and concrete implementations that the Slice compiler generates from
your Slice interfaces.

{% callout %}

> An interface is a Slice construct but not a Slice type. In particular, you cannot use an interface as the type of a
> Slice field or operation parameter.

Reviewer: ```suggestion
An interface is a Slice construct but not a Slice type. This means you cannot use an interface as the type of a
Slice field or operation parameter.
```

I think its a bit clearer when phrased like this.
